MediaGalleryRenditions/Model/RenditionsImageManagement.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryRenditions\Model;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\Framework\Filesystem\Io\File;
+
+class RenditionsImageManagement
+{
+    /**
+     * @var Filesystem\Directory\WriteInterface
+     */
+    private $_directory;
+
+    /**
+     * @var File|mixed|null
+     */
+    private $ioFile;
+
+    /**
+     * @var DriverInterface|mixed|null
+     */
+    private $file;
+
+    private const RENDITIONS_DIRECTORY_NAME = '.renditions';
+
+    /**
+     * @param Filesystem $filesystem
+     * @param File|null $ioFile
+     * @param DriverInterface|null $file
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $filesystem,
+        File $ioFile = null,
+        DriverInterface $file = null
+    ) {
+        $this->_directory = $filesystem->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->ioFile = $ioFile ?: ObjectManager::getInstance()->get(File::class);
+        $this->file = $file ?: ObjectManager::getInstance()->get(Filesystem\Driver\File::class);
+    }
+
+    /**
+     * Returns Rendition image path
+     *
+     * @param string $path
+     * @return string
+     * @throws LocalizedException
+     * @throws FileSystemException
+     */
+    public function execute(string $path) :string
+    {
+        $realPath = $this->_directory->getRelativePath($path);
+        if (!$this->_directory->isFile($realPath) || !$this->_directory->isExist($realPath)) {
+            throw new LocalizedException(__('Directory or File %1 does not exist in media directory.', $realPath));
+        }
+        $renditionImageDirectoryPath = $this->getRenditionsImageDirectory($path);
+        $renditionImageDirectory = $this->_directory->getRelativePath($renditionImageDirectoryPath);
+        if (!$this->_directory->isExist($renditionImageDirectory)) {
+            $this->_directory->create($renditionImageDirectory);
```

Comment:
Let's create the directory only when creating a rendition. Let's just return the constructed rendition path from this service and do not check if directory or file exists